File: python/src/daily_forecasts.py
import datetime

import scorer
from scorer import IceCream


def print_sales_forecasts(now=None):
    def update_selection():
        scorer.update_selection()

    print_sales_forecase_and_update_selection(now, update_selection)


def print_sales_forecase_and_update_selection(now, update_selection):
    names = ["Steve", "Julie", "Francis"]
    now = now or datetime.datetime.now()
    print(f"Forecast at time {now}")
    for name in names:
        if name == "Steve":
            scorer.flavour = IceCream.Strawberry
        else:
            update_selection()
        score = scorer.get_sales_forecast()
        print(f"{name} score: {score}")

File: csharp/src/DailyForecasts.cs
using System;
using System.Collections.Generic;

namespace IceCreamScorer
{
    public class DailyForecasts
    {
        private readonly Scorer scorer;
        public DailyForecasts(Scorer scorer)
        {
            this.scorer = scorer;
        }

        public void PrintSalesForecasts()
        {
            var names = new List<string>() { "Steve", "Julie", "Francis" };
            Console.WriteLine($"Scoreboard at time: {DateTime.Now:ddd MMM dd hh:mm:ss yyyy}");

            foreach (var name in names)
            {
                if (name == "Steve")
                {
                    scorer.Flavour = IceCream.Strawberry;
                }
                else
                {
                    scorer.UpdateSelection();
                }
                int score = scorer.GetScore();
                Console.WriteLine($"{name} score: {score}");
            }
        }
    }
}
